packages/amplify-cli/src/app-config/config.ts

```diff
@@ -0,0 +1,52 @@
+import fs from 'fs-extra';
+import uuid from 'uuid';
+import _ from 'lodash';
+
+import { Context } from '../domain/context';
+import { getPath } from './getPath';
+
+export function init(context: Context) {
+  const configPath = getPath(context);
+  if (fs.existsSync(configPath)) {
+    Config.Instance.setValues(fs.readFileSync(configPath));
+  } else {
+    write(context, Config.Instance);
+  }
+  return getConfig();
+}
+
+export function getConfig() {
+  return Config.Instance;
+}
+
+export function write(context: Context, keyValues: Object) {
+  Config.Instance.setValues(keyValues);
+  fs.writeFileSync(getPath(context), JSON.stringify(Config.Instance));
```

Comment:
I think this can fail if `~/.amplify` directory does not exist, not?

Reply:
~/.amplify the directory is created during a plugin scan.